src/device/cli.ts

```diff
@@ -2,59 +2,543 @@
+  const dataVolume = data.withVolume(docker, await data.volume(docker))
+  const device = await dataVolume.read()
+
+  const toPrint: Record<string, string> = {
+    Network: toUpperCaseFirst(device.network),
+    Device: device.address
+  }
+
+  try {
+    const address = await withFile(opts.wallet).address()
+    const stakes = await xe.stake.stakes(network.blockchain.baseURL, address)
+    const stake = Object.values(stakes).find(s => s.device === device.address)
+    if (stake !== undefined) {
+      toPrint.Type = toUpperCaseFirst(stake.type)
+      toPrint.Stake = printID(stake.id)
+    }
+    else toPrint.Stake = 'Unassigned'
+  }
+  catch (err) {
+    toPrint.Stake = 'Unassigned (no wallet)'
+  }
+
+  console.log(printData(toPrint))
+}
+
+const infoHelp = [
+  '\n',
+  'This command displays information about your device and the stake it is assigned to.'
+].join('')
+
+const removeAction = (parent: Command, removeCmd: Command, network: Network) => async () => {
+  const opts = {
+    ...walletCLI.getWalletOption(parent, network),
+    ...walletCLI.getPassphraseOption(removeCmd),
+    ...(() => {
+      const { fullIds, yes } = removeCmd.opts<{ fullIds: boolean, yes: boolean }>()
+      return { fullIds, yes }
+    })()
+  }
+  const printID = printTrunc(!opts.fullIds, 8)
+
+  const storage = withFile(opts.wallet)
+  const docker = new Docker(getDockerOptions(removeCmd))
+
+  const dataVolume = data.withVolume(docker, await data.volume(docker))
+  const device = await dataVolume.read()
+
+  const stakes = await xe.stake.stakes(network.blockchain.baseURL, await storage.address())
+  const stake = Object.values(stakes).find(s => s.device === device.address)
+  const nodeName = stake !== undefined ? toUpperCaseFirst(stake.type) : ''
+
+  // confirm user intent
+  if (!opts.yes) {
+    console.log(`You are removing this device from Edge ${toUpperCaseFirst(network.name)}.`)
+    console.log()
+    if (stake === undefined) console.log('This device is not assigned to any stake.')
+    else console.log(`This will remove this device's assignment to stake ${printID(stake.id)} (${nodeName}).`)
+    console.log()
+    let confirm = ''
+    const ynRegexp = /^[yn]$/
+    while (confirm.length === 0) {
+      const input = await ask('Remove this device? [yn] ')
+      if (ynRegexp.test(input)) confirm = input
+      else console.log('Please enter y or n.')
+    }
+    if (confirm === 'n') return
+    console.log()
+  }
+
+  if (stake !== undefined) {
+    // if required, create unassignment tx
+    await askToSignTx(opts)
+    const wallet = await storage.read(opts.passphrase as string)
+    const api = xeWithNetwork(network)
+    const onChainWallet = await api.walletWithNextNonce(wallet.address)
+
+    const tx = xe.tx.sign({
+      timestamp: Date.now(),
+      sender: wallet.address,
+      recipient: wallet.address,
+      amount: 0,
+      data: {
+        action: 'unassign_device',
+        memo: 'Unassign device from stake',
```

Comment:
As per brand guidelines, this should be title case. I recommend the memo `Unassign Device`.